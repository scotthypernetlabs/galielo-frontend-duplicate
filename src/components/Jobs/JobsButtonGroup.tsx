import {ToggleButton, ToggleButtonGroup} from "@material-ui/lab";
import React from "react";
import {createStyles, WithStyles, withStyles} from "@material-ui/core";
import {galileoDarkBlue} from "../theme";

interface Props extends WithStyles<typeof styles>{
  toggleMode: any;
  mode: boolean;
}

type State = {}

const styles = () => createStyles({
  root: {
    minWidth: 175,
    width: "100%"
  },
  selected: {
    backgroundColor: galileoDarkBlue.main,
    color: "white",
    // "&:hover": {
    //   backgroundColor: galileoDarkBlue.main
    // }
  }
});

class JobsButtonGroup extends React.Component<Props, State> {
  constructor(props: Props){
    super(props)
  }

  render() {
    const {classes} = this.props;
    return(
      <ToggleButtonGroup
      >
        <ToggleButton
          classes={{root: classes.root, selected: classes.selected}}
          value="Sent"
          selected={this.props.mode}
          onClick={this.props.toggleMode}
        >
          Sent
        </ToggleButton>
        <ToggleButton
          classes={{root: classes.root}}
          value="Received"
          selected={!this.props.mode}
          onClick={this.props.toggleMode}
        >
          Received
        </ToggleButton>
      </ToggleButtonGroup>
    )
  }
}

export default withStyles(styles)(JobsButtonGroup)
