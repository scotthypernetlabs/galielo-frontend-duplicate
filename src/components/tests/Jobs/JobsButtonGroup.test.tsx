import { mount } from "enzyme";
import JobsButtonGroup from "../../Jobs/JobsButtonGroup";
import React from "react";
import ToggleButton from "@material-ui/lab/ToggleButton";

describe("JobsButtonGroup Component", () => {
  const componentProps = {
    toggleMode: jest.fn(),
    mode: true
  };

  const Wrapper = (props: any) => {
    return <JobsButtonGroup {...componentProps} />;
  };

  const jobsButtonGroup = mount(<Wrapper />);

  it("buttons should have the right text", () => {
    expect(jobsButtonGroup.find(ToggleButton)).toHaveLength(2);
    expect(
      jobsButtonGroup
        .find(ToggleButton)
        .at(0)
        .childAt(0)
        .text()
    ).toBe("Sent");
    expect(
      jobsButtonGroup
        .find(ToggleButton)
        .at(1)
        .childAt(0)
        .text()
    ).toBe("Received");
  });

  it("buttons should call toggleMode", () => {
    jobsButtonGroup
      .find(ToggleButton)
      .at(0)
      .simulate("click");
    expect(jobsButtonGroup.childAt(0).props().toggleMode).toHaveBeenCalled();
    jobsButtonGroup
      .find(ToggleButton)
      .at(1)
      .simulate("click");
    expect(jobsButtonGroup.childAt(0).props().toggleMode).toHaveBeenCalled();
  });

  it("should match snapshot", () => {
    expect(jobsButtonGroup).toMatchSnapshot();
  });
});
