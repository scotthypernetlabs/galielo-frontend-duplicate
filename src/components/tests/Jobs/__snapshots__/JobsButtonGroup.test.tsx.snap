// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsButtonGroup Component should match snapshot 1`] = `ReactWrapper {}`;
